src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -1102,6 +1105,45 @@ void Transformations::MainSnippets(void) {
     },
     snippets::pass::TokenizeSnippets);
 
+    auto mm_supports_transpose_b = [this](const std::shared_ptr<const ov::Node>& n) {
+        MAYBE_UNUSED(inferencePrecision);
+        const auto& b_shape = n->get_input_partial_shape(1);
+        if (b_shape.is_dynamic())
+            return false;
+        // Note: BrgemmTPP doesn't support transposed KN natively
+        // so we should extract transposes for the corresponding matmul nodes
+#if defined(SNIPPETS_LIBXSMM_TPP)
+        std::vector<std::vector<size_t>> layouts(3);
+        const auto matmul = ov::as_type_ptr<const ov::op::v0::MatMul>(n);
+        OPENVINO_ASSERT(matmul, "ExplicitTransposeMatMulInputs callback must be called for matmul node");
+        if (matmul->get_transpose_b()) {
+            std::vector<size_t> transposed_layout(b_shape.size());
+            std::iota(transposed_layout.begin(), transposed_layout.end(), 0);
+            std::swap(*transposed_layout.rbegin(), *(transposed_layout.rbegin() + 1));
+            layouts[1] = std::move(transposed_layout);
+        }
+        ov::element::TypeVector precisions;
+        auto push_precision = [&](const ov::element::Type& precision) {
+            if (inferencePrecision == ov::element::bf16 && precision == ov::element::f32)
+                precisions.push_back(ov::element::bf16);
+            else
+                precisions.push_back(precision);
+        };
+        push_precision(n->get_input_element_type(0));
+        push_precision(n->get_input_element_type(1));
+        push_precision(n->get_output_element_type(0));
+        if (ov::intel_cpu::tpp::pass::BrgemmToBrgemmTPP::is_supported_brgemm_configuration(layouts, precisions))
+            return false;
```

Comment:
The logic here is following: the TPP implementation **doesn't support** transpose_b=true, so if we know that the MM will use TPP impl, we return false from `mm_supports_transpose_b` in order to extract transpose from MM.
If TPP doesn't support such MM, it will be converted to BrgemmCPU, which supports transpose_b=true, so we return true is such cases.

Reply:
Thank you for explanation! Very deep function call to understand it at first time 😄